Looks like there's a build issue:

```
PASS Bug #80763 (msgfmt_format() does not accept DateTime references) [tests/bug80763.phpt] 
gcc: warning:  -DZEND_COMPILE_DL_EXT=1: linker input file unused because linking not done
gcc: error:  -DZEND_COMPILE_DL_EXT=1: linker input file not found: No such file or directory
make: *** [Makefile:206: apc.lo] Error 1
checking pkg-config is at least version 0.9.0... yes
error: builder for '/nix/store/dh7rkk42hyjb48c41yimyyr6cy0y8x1x-php-apcu-5.1.23.drv' failed with exit code 2;
       last 10 log lines:
       > configure: creating ./config.status
       > config.status: creating config.h
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash EXTENSION_DIR=\$\(out\)/lib/php/extensions phpincludedir=\$\(dev\)/include
       > /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/source/libtool --tag=CC --mode=compile gcc -I. -I/build/source -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/main -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/TSRM -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/Zend -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/ext -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE   \ -DZEND_COMPILE_DL_EXT=1 -c /build/source/apc.c -o apc.lo  -MMD -MF apc.dep -MT apc.lo
       > mkdir .libs
       >  gcc -I. -I/build/source -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/main -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/TSRM -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/Zend -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/ext -I/nix/store/lkcbqnwg5sh196yjiry3h6nwnvbwyqgf-php-8.4.1-dev/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE " -DZEND_COMPILE_DL_EXT=1" -c /build/source/apc.c -MMD -MF apc.dep -MT apc.lo  -fPIC -DPIC -o .libs/apc.o
       > gcc: warning:  -DZEND_COMPILE_DL_EXT=1: linker input file unused because linking not done
       > gcc: error:  -DZEND_COMPILE_DL_EXT=1: linker input file not found: No such file or directory
       > make: *** [Makefile:206: apc.lo] Error 1
       For full logs, run 'nix-store -l /nix/store/dh7rkk42hyjb48c41yimyyr6cy0y8x1x-php-apcu-5.1.23.drv'.
```

--- a/pkgs/development/php-packages/apcu/default.nix
+++ b/pkgs/development/php-packages/apcu/default.nix
@@ -7,7 +7,7 @@
 }:
 
 let
-  version = "5.1.23";
+  version = "5.1.24";
 in
 buildPecl {
   inherit version;
@@ -17,24 +17,8 @@ buildPecl {
     owner = "krakjoe";
     repo = "apcu";
     rev = "v${version}";
-    sha256 = "sha256-UDKLLCCnYJj/lCD8ZkkDf2WYZMoIbcP75+0/IXo4vdQ=";
+    sha256 = "sha256-g+Oo6y+24VOWHaDZ23iItkGwOa5bTtKMAjZOmAi6EOo=";
   };
-
-  patches = [
-    # Fix broken test (apc_entry_002) with PHP 8.4 alpha1
-    # See https://github.com/krakjoe/apcu/issues/510
-    (fetchpatch {
-      url = "https://github.com/krakjoe/apcu/commit/9dad016db50cc46321afec592ea9b49520c1cf13.patch";
-      hash = "sha256-8CPUNhEGCVVSXWYridN1+4N4JzCfXZbmUIsPYs/9jfk=";
-    })
-
-    # Fix ZTS detection in tests with PHP 8.4
-    # https://github.com/krakjoe/apcu/pull/511
-    (fetchpatch {
-      url = "https://github.com/krakjoe/apcu/commit/15766e615264620427c2db37061ca9614d3b7319.patch";
-      hash = "sha256-gbSkx47Uo9E28CfJJj4+3ydcw8cXW9NNN/3FuYYTVPY=";
-    })
-  ];
 
   buildInputs = [ pcre2 ];
   doCheck = true;

